src/GovUk.Education.ExploreEducationStatistics.Common.Tests/Extensions/AssertExtensions.cs

```diff
@@ -10,24 +14,58 @@ public static class AssertExtensions
          * Calling this method causes a Test to fail with the given message.  The equivalent of `Assert.Fail()` in
          * other testing frameworks.
          */
-        public static XunitException AssertFail(string message) 
+        public static XunitException AssertFail(string message)
         {
             throw new XunitException(message);
         }
 
-        public static bool AssertDeepEqualTo<T>(this T actual, T expected)
+        public static bool AssertDeepEqualTo<T>(
+            this T actual,
+            T expected,
+            Expression<Func<T, object>>[]? ignoreProperties = null)
         {
             var compareLogic = new CompareLogic();
+            ignoreProperties?.ForEach(compareLogic.Config.IgnoreProperty);
             var comparison = compareLogic.Compare(actual, expected);
             Assert.True(comparison.AreEqual, comparison.DifferencesString);
             return true;
         }
-        
-        public static bool IsDeepEqualTo<T>(this T actual, T expected)
+
+        /// <summary>
+        /// A convenience method for combining AssertDeepEqualTo with one or more ignorable fields, to exclude certain
+        /// fields from the deep equality check.
+        /// </summary>
+        public static Expression<Func<T, object>>[] Ignoring<T>(params Expression<Func<T, object>>[] properties)
+        {
+            return properties;
+        }
+
+        public static bool IsDeepEqualTo<T>(
+            this T actual,
+            T expected,
+            Expression<Func<T, object>>[]? ignoreProperties = null)
         {
             var compareLogic = new CompareLogic();
+            ignoreProperties?.ForEach(compareLogic.Config.IgnoreProperty);
             var comparison = compareLogic.Compare(actual, expected);
             return comparison.AreEqual;
         }
+
+        /// <summary>
+        /// Assert that the given DateTime is effectively "now", within a given tolerance of milliseconds.
+        /// </summary>
+        public static void AssertUtcNow(this DateTime dateTime, int withinMillis = 1500)
```

Comment:
Is the tolerance a bit on the high side? Guess we want to avoid false negatives due to timing issues, but being 1.5 seconds out is potentially a bit much? 

Maybe 1 second at most would be more appropriate?

Reply:
Fine with this, yeah.  I only kept it at 1500 because that was the example used everywhere around the codebase already.  I've pushed the boat out and set it to 500ms, which should be plenty.

Reply:
I've also added in additional tests for the nullable DateTime AssertUtcNow override.